lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts

```diff
@@ -0,0 +1,303 @@
+import { CachedRoutes, CacheMode, ChainId, IRouteCachingProvider, log } from '@uniswap/smart-order-router'
+import { DynamoDB } from 'aws-sdk'
+import { Currency, CurrencyAmount, Token, TradeType } from '@uniswap/sdk-core'
+import { Protocol } from '@uniswap/router-sdk'
+import { CACHED_ROUTES_CONFIGURATION } from './cached-routes-configuration'
+import { PairTradeTypeChainId } from './model/pair-trade-type-chain-id'
+import { CachedRoutesMarshaller } from './marshalling/cached-routes-marshaller'
+import { CachedRoutesStrategy } from './model/cached-routes-strategy'
+
+interface ConstructorParams {
+  /**
+   * The TableName for the DynamoDB Table. This is wired in from the CDK definition.
+   */
+  cachedRoutesTableName: string
+  /**
+   * The amount of minutes that a CachedRoute should live in the database.
+   * This is used to limit the database growth, Dynamo will automatically delete expired entries.
+   */
+  ttlMinutes?: number
+}
+
+export class DynamoRouteCachingProvider extends IRouteCachingProvider {
+  private readonly ddbClient: DynamoDB.DocumentClient
+  private readonly tableName: string
+  private readonly ttlMinutes: number
+
+  constructor({ cachedRoutesTableName, ttlMinutes = 5 }: ConstructorParams) {
+    super()
+    this.ddbClient = new DynamoDB.DocumentClient()
```

Comment:
I think the default retry strategy for AWS SDK calls is 3 retries with exponential backoff. Not sure what the default timeout is

Wondering if this is the best config for a cache? May want to fast fail with 0 retries? And timeout at 100ms or something?

Reply:
Updated :)